lib/dao.js

```diff
@@ -757,12 +757,12 @@ DataAccessObject.findOrCreate = function findOrCreate(query, data, options, cb)
 
         // options has precedence on model-setting
         if (options.validate === false) {
-          _findOrCreate(query, data, obj);
+          return _findOrCreate(query, data, obj);
```

Comment:
Hmm, I got confused here. Since this pull request is changing findOrCreate and we need to back-port it to 2.x in whole, it's ok to include related fixes. Just make sure the fix is in a standalone commit and includes unit-tests.